### PR TITLE
Add TagFilterButton plugin

### PR DESCRIPTION
**Pull request by Waffleboy**
A very useful (and even basic) feature for interactive maps is dynamic filtering of points. 

There's a plugin for leaflet called **Tag Filter Button** which does precisely that, add a button that will dynamically filter the maps based on a feature.

[Demo available here](http://maydemirx.github.io/leaflet-tag-filter-button/)

[Link to plugin](https://github.com/maydemirx/leaflet-tag-filter-button)

I have built the plugin for folium, ~sans the test and examples~. If this is chosen to be added, I'll add them as well

EDIT: Tests and examples added as well.

### CODE REVIEW

File: folium/plugins/tag_filter_button.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        {% endmacro %}
+        """)
+
+    def __init__(self, data, name=None, icon="fa-filter",
+                 on_selection_complete=None,
+                 clear_text='clear', filter_on_every_click=True,
+                 open_popup_on_hover=False, 
+                 overlay=True, control=True, show=True, **kwargs):
+        super(TagFilterButton, self).__init__(name=name, overlay=overlay,
+                                      control=control, show=show)
```
[the remark this answers is not in the document]
Comment (Waffleboy) [reply folium/plugins/tag_filter_button.py]: Fixed

File: folium/plugins/tag_filter_button.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    def render(self, **kwargs):
+        super(TagFilterButton, self).render(**kwargs)
+
+        figure = self.get_root()
+        assert isinstance(figure, Figure), ('You cannot render this Element '
+                                            'if it is not in a Figure.')
+
+        figure.header.add_child(
+            JavascriptLink('https://cdn.jsdelivr.net/npm/leaflet-tag-filter-button@0.0.4/src/leaflet-tag-filter-button.js'),
```
[the remark this answers is not in the document]
Comment (Waffleboy) [reply folium/plugins/tag_filter_button.py]: Fixed

File: folium/plugins/tag_filter_button.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+            name='tag-filter-button.js')
+            # JavascriptLink('https://raw.githubusercontent.com/maydemirx/leaflet-tag-filter-button/master/src/leaflet-tag-filter-button.js'),  # noqa
+
+        figure.header.add_child(
+            JavascriptLink('https://cdn.jsdelivr.net/npm/leaflet-easybutton@2/src/easy-button.js'),  # noqa
+            name='easy-button.js')
+
```
[the remark this answers is not in the document]
Comment (Waffleboy) [reply folium/plugins/tag_filter_button.py]: Fixed

File: folium/plugins/tag_filter_button.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        
+        figure.header.add_child(
+            CssLink('https://cdn.jsdelivr.net/npm/leaflet-tag-filter-button@0.0.4/src/leaflet-tag-filter-button.css'),
+            name='tag-filter-button.css')
+        # CssLink('https://raw.githubusercontent.com/maydemirx/leaflet-tag-filter-button/master/src/leaflet-tag-filter-button.css'),  # noqa
+
+        figure.header.add_child(
+            CssLink('https://cdn.jsdelivr.net/npm/leaflet-easybutton@2/src/easy-button.css'),
+            name='easy-button.css')
+
```
[the remark this answers is not in the document]
Comment (Waffleboy) [reply folium/plugins/tag_filter_button.py]: Fixed

File: folium/plugins/tag_filter_button.py [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+        # CssLink('https://raw.githubusercontent.com/maydemirx/leaflet-tag-filter-button/master/src/leaflet-tag-filter-button.css'),  # noqa
+
+        figure.header.add_child(
+            CssLink('https://cdn.jsdelivr.net/npm/leaflet-easybutton@2/src/easy-button.css'),
+            name='easy-button.css')
+        
+        figure.header.add_child(
+            CssLink('https://cdn.jsdelivr.net/npm/css-ripple-effect@1.0.5/dist/ripple.min.css'),
+            name='ripples.min.css')
+
```
[the remark this answers is not in the document]
Comment (Waffleboy) [reply folium/plugins/tag_filter_button.py]: Fixed

**Comment by Waffleboy** [comment]
Added Examples and Tests.

**Comment by hannahrosey** [comment]
Thank you for building this!!

**Review by Conengmo — APPROVED** [review]
Good work! I updated the PR based on some improvements that came available later.

I removed the version for the plugin CDN link, so we'll always use the latest version.

I removed the `onSelectionComplete` argument, since this refers to a Javascript function. This might confuse our Python users. Power users can still pass this through the `kwargs`.

Let's merge this when the tests pass.

**Comment by Conengmo** [comment]
Thank you @Waffleboy for your PR! Sorry it took so long to get processed. It will be included in the next folium release, 0.14.0.

**Pull request merged**